python/agents/camel/camel/camel_library/capabilities/utils.py

```
@@ -0,0 +1,149 @@
+
+def _source_is_trusted(s: sources.Source, trusted_set: set[Any] | None) -> bool:
+  trusted_set = trusted_set or _TRUSTED_SET
+  if s in trusted_set:
+    return True
+  if (
+      isinstance(s, sources.Tool)
+      and not s.inner_sources
+      and s.inner_sources.issubset(_TRUSTED_SET)
+  ):
+    return True
+  return False
```

Ruff (0.12.2)

Return the condition directly

Inline condition

(SIM103)